lvsfunc/misc.py

```diff
@@ -469,6 +472,123 @@ def colored_clips(amount: int,
+                          as the script is ran in) (Default: ``None``\ )
+    """
+    @contextlib.contextmanager
+    def _cd(newdir):
+        prevdir = os.getcwd()
+        os.chdir(os.path.expanduser(newdir))
+        try:
+            yield
+        finally:
+            os.chdir(prevdir)
+
+    if 0 in (clip.height for clip in clips.values()) or 0 in (clip.width for clip in clips.values()):
+        raise ValueError("save: variable-resolution clips not allowed")
+    if None in (clip.format for clip in clips.values()):
+        raise ValueError("save: variable-format clips not allowed")
+
+    if save_location is not None:
+        if (save_location := pathlib.Path(save_location)).exists():
+            os.chdir(save_location)
+
+    if len(clips) > 1:
+        if len(set(clip.height for clip in clips.values())) > 1 or len(set(clip.width for clip in clips.values())) > 1:
+            raise ValueError("save: all clips must have same dimensions")
+
+    max_frame = min(clip.num_frames for clip in clips.values()) - 1
+    if not all(f < max_frame for f in frames):
+        raise ValueError("save: specified frame numbers out of range on one or more clips")
+    if len(frames) == 0 and random_number == 0:
+        random_number = 1
+
+    if random_number := abs(random_number):
+        if random_number == 1:
+            while (rand_frame_no := random.randint(0, max_frame)) in frames:  # prevents adding the same random frame number as one specified in `frames`
+                pass
+            frames.append(rand_frame_no)
+        else:
+            random_frame_numbers = random.sample(range(max_frame), random_number)
+            while not all(i not in frames for i in random_frame_numbers):
+                random_frame_numbers = random.sample(range(max_frame), random_number)
+            frames += random_frame_numbers
+
+    for name, clip in clips.items():
+        range_in = Range.LIMITED if clip.format.color_family == vs.ColorFamily.YUV else Range.FULL
+        clips[name] = depth(clip, 8, range_in=range_in, dither_type=dither_type)
+        if (subsampling := get_subsampling(clips[name])) and subsampling not in ('444', None):
+            clips[name] = clips[name].resize.Bicubic(filter_param_a_uv=1/3,
+                                                     filter_param_b_uv=1/3,
+                                                     format=clip.format.replace(subsampling_w=0, subsampling_h=0)
+                                                     )
+
+    if folder:
+        for name, clip in clips.items():
+            if clip.format.color_family == vs.ColorFamily.YUV:
+                matrix_in_s = '709'
+            elif clip.format.color_family == vs.ColorFamily.RGB:
+                matrix_in_s = 'rgb'
+            else:
+                raise ValueError("save: expected a YUV or RGB clip")
+            os.makedirs(name, exist_ok=True)
+            with _cd(name):
```

Comment:
Is this required to work around a limitation in `core.imwri.Write`'s API? If yes, you should add a comment for that. If no, remove it and do it like in the `folder==False` branch?